### ISSUE TITLE
508-defect-2 [FOCUS]: Focus isn't managed for the list loop other names pattern

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#43262]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader and keyboard user, I expect focus to be managed when interacting with dynamic fields that appear and disappear on the page.

## Details

Currently the other name list loop pattern fails to manage focus when:
- Another name is added. Focus is not managed and remains on the add another name button.
- A name is removed. Focus is lost.
- A name is updated. Focus is lost.
- A name is edited. Focus is sent to the h1.

## Acceptance Criteria

- [ ] Focus is sent to the first new field when "Add another name" is clicked
- [ ] Focus is returned to the "Add another name" button when the remove button is clicked
- [ ] Focus is sent to the first new field when "Edit" is clicked
- [ ] Focus is returned to "Edit" when "Update" is clicked

## Steps to Recreate

1. [Go to the "other name" page on step 3](https://www.va.gov/burials-and-memorials/application/530/military-history/service-periods)
2. Click "yes"
3. Fill in the fields and use `space` to click "Add another name"
4. Confirm focus remains on the add another name button
5. Shift tab to remove and click using `space`
6. Confirm focus is not sent back to the add another name button
7. Tab to add another name and click using `space`
8. Shift tab back to edit and click using `space`
9. Confirm focus is not sent to the first field
10. Tab to the update button and click using `space`
11. Confirm focus is not sent back to the `edit` field

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/175342508-622c16ab-c66b-4aa9-bb3f-202058077e46.mov

**Comment by jacobworrell** [comment]
@joshkimux after talking with @jerekshoe it seems like this is an issue that would need to be addressed for every form on VA.gov and not something we could isolate to just our applications. We'll need consensus on a broader solution so considering this blocked for now?

**Comment by obliviga** [comment]
I'm looking into this, hoping to provide some quick fixes.

**Comment by joshkimux** [comment]
@obliviga , @Mottie is a great person to talk to about this! FYI, I just heard today that @Mottie (Robin) and Pam are working on an experimental design system component to fix all of this. That might be worth checking in on too 😄
